eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -139,21 +157,42 @@ stages:
                           
                           Write-Host "git status"
                           git status
-                        displayName: Copy updated files into vcpkg fork for PR
-                        condition: ne(variables['SkipVcpkgUpdate'], 'true')
 
-                      # This template does not need to check the 
-                      # SkipVcpkgUpdate variable, it will stop if there are no
-                      # changes in the git repo.
+                          # Commit changes
+                          Write-Host "git add -A"
+                          git add -A 
+                          Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" commit -m `"Initial vcpkg commit for ${{ artifact.VcpkgPortName }}`""
+                          git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit  -m "Initial vcpkg commit for ${{ artifact.VcpkgPortName }}" 
+
+                          # Run vcpkg x-add-version to add version to vcpkg file
+                          Write-Host "./bootstrap-vcpkg.bat"
+                          ./bootstrap-vcpkg.bat
+                          Write-Host "./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}"
+                          ./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}
+
+                          # Amend commit to include results of x-add-version
+                          Write-Host "git status"
+                          git status
+                          Write-Host "git add -A"
+                          git add -A
+                          Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" commit --amend -m `"[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)`""
+                          git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit --amend -m "[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)"
+
+                          # Work around create-pull-request.yml logic that 
+                          # checks for changes
```

Comment:
Why commit this and then override the HasChanges? Can't we let the create-pull-request template do the commit?